frontend/views/containers/chatroom/MessageBase.vue

```diff
@@ -236,6 +298,47 @@ export default ({
       // Extract only numbers and colons, removing non-numeric characters like AM/PM
       const cleaned = tString.replace(/[^\d:]/g, '')
       return cleaned
+    },
+    determineToEnableTruncationToggle () {
+      const msgBodyEl = this.isTypePoll
+        ? this.$refs.msgFullWidthBody // poll message is rendered in .c-full-width-body container
+        : this.$refs.msgBody
+      const threshold = this.chatMainConfig.isPhone
+        ? CHAT_LONG_MESSAGE_HEIGHT_THRESHOLD_MOBILE
+        : CHAT_LONG_MESSAGE_HEIGHT_THRESHOLD_DESKTOP
+
+      if (msgBodyEl) {
+        // If the truncate-toggle is already displayed and the message is folded,
+        // we should use scrollHeight value for the check instead of clientHeight.
+        const height = this.isMessageCropped ? msgBodyEl.scrollHeight : msgBodyEl.clientHeight
+        this.ephemeral.truncateToggle.enabled = height > threshold
+      }
+    },
+    toggleMessageTruncation (e) {
+      const currentValue = this.ephemeral.truncateToggle.isShowingAll
+      this.ephemeral.truncateToggle.isShowingAll = !currentValue
+
+      if (currentValue) {
+        // When folding the expanded message, scroll to the message so that it stays in the screen.
+        this.$el.scrollIntoView({ behavior: 'instant' })
+      }
+    }
+  },
+  watch: {
+    'chatMainConfig.isPhone' () {
+      if (this.shouldCheckToTruncate) {
+        this.determineToEnableTruncationToggle()
+      }
+    }
+  },
+  mounted () {
+    if (
+      this.shouldCheckToTruncate &&
+      // NOTE: If the message has any image attached, defer this check until the <img /> DOMs are rendered.
+      //       (which is detected via 'image-attachments-render-complete' custom event in ChatAttachmentPreview.vue)
+      !this.hasImageAttachment
+    ) {
+      this.determineToEnableTruncationToggle()
```

Comment:
The "Show more" toggle is determined once when the message is loaded. However, if the message is edited, `determineToEnableTruncationToggle` isn't calculated again.

So if the message is short initially, and then edited to be super long, `Show more` won't appear.

Likewise, if the message is long initially, and then made short, `Show more` won't disappear.

So maybe another watcher or something needs to be added for updating on message edit.

Reply:
@taoeffect Have implemented this requirement by adding a watcher for `updatedDate` property of the message. By doing so, we can guarantee that the height calculation/comparison is run 'after' the updated message is already reflected(or rendered) to the DOM.